internal/worker/queue_test.go

```diff
@@ -135,60 +158,149 @@ func Test_queue_Start(t *testing.T) {
 		if !errors.Is(err, w.err) {
 			return errors.Errorf("got error = %v, want %v", err, w.err)
 		}
-		if !reflect.DeepEqual(got, w.want) {
+		opts := []cmp.Option{
+			cmp.Comparer(func(want, got <-chan error) bool {
+				return (want != nil && got != nil) || (want == nil && got == nil)
```

Comment:
do we need to check the error sent to channel?

Reply:
maybe you can reference this implementation.
https://github.com/vdaas/vald/blob/test/internal/worker/internal/worker/worker_test.go#L217

Reply:
> do we need to check the error sent to channel?

I think so too:pray:
we want to check error into the channel :pray:
for examp, please take out the errors from the want and got channels and compare them. 🤔